src/datasources/balances-api/coingecko-api.service.ts

```diff
@@ -92,12 +100,12 @@ export class CoingeckoApi implements IPricesApi {
         notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
         expireTimeSeconds: this.pricesTtlSeconds,
       });
+      return result?.[nativeCoinId]?.[lowerCaseFiatCode];
     } catch (error) {
-      throw new DataSourceError(
-        `Error getting ${
-          args.nativeCoinId
-        } price from provider${this._mapProviderError(error)}`,
+      this.loggingService.error(
+        `CoinGecko error while getting native coin price: ${asError(error)} `,
       );
+      return null;
```

Comment:
As above, these `DataSourceError`s were cached by `PricesRepository`, which mapped them to `null`. This logic is also pushed here, hopefully this simplifies the overall flow.

Reply:
Nit: can you add a comment regarding this?